api/types/constants.go

```diff
@@ -134,6 +134,9 @@ const (
 	// KindKubernetesCluster is a Kubernetes cluster.
 	KindKubernetesCluster = "kube_cluster"
 
+	// KindKubePod is an Kubernetes Pod resource type.
+	KindKubePod = "pod"
```

Comment:
I think it's easier to use the string since you can create a `RoleV6` with  the following spec:

```yaml
kind: role
version: v6
metadata:
  name: my-kube-role
spec:
  allow:
    kubernetes_labels:
      "*": "*"
    kubernetes_groups: ["kube_group"]
    kubernetes_resources:
    - kind: pod
      name: A
      namespace: namespace
    - kind: pod
      name: B
      namespace: namespace
    - kind: pod
      name: C
      namespace: other-namespace
```

If kind is an enum we need to convert it from string to the enum value (if the underlying enum is a string, we do not need to do anything) but we still need to validate it since we accept the value from outside